crates/turbopack-css/src/chunk/source_map.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::Result;
+use turbo_tasks::{primitives::StringVc, ValueToString, ValueToStringVc};
+use turbo_tasks_fs::{File, FileSystemPathVc};
+use turbopack_core::{
+    asset::{Asset, AssetContentVc, AssetVc},
+    reference::{AssetReference, AssetReferenceVc, AssetReferencesVc},
+    resolve::{ResolveResult, ResolveResultVc},
+    source_map::GenerateSourceMap,
+};
+
+use super::CssChunkVc;
+
+/// Represents the source map of an css chunk.
+#[turbo_tasks::value]
+pub struct CssChunkSourceMapAsset {
+    chunk: CssChunkVc,
+}
+
+#[turbo_tasks::value_impl]
+impl CssChunkSourceMapAssetVc {
+    #[turbo_tasks::function]
+    pub fn new(chunk: CssChunkVc) -> Self {
+        CssChunkSourceMapAsset { chunk }.cell()
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl Asset for CssChunkSourceMapAsset {
+    #[turbo_tasks::function]
+    async fn path(&self) -> Result<FileSystemPathVc> {
+        Ok(self.chunk.path().append(".map"))
+    }
+
+    #[turbo_tasks::function]
+    async fn content(&self) -> Result<AssetContentVc> {
+        let sm = self.chunk.generate_source_map().to_rope().await?;
+        Ok(File::from(sm).into())
+    }
+
+    #[turbo_tasks::function]
+    fn references(&self) -> AssetReferencesVc {
+        AssetReferencesVc::empty()
+    }
+}
+
+/// A reference to a [`CssChunkSourceMapAsset`], used to inform the dev
+/// server/build system of the presence of the source map
+#[turbo_tasks::value]
+pub struct CssChunkSourceMapAssetReference {
+    chunk: CssChunkVc,
+}
+
+#[turbo_tasks::value_impl]
+impl CssChunkSourceMapAssetReferenceVc {
+    #[turbo_tasks::function]
+    pub fn new(chunk: CssChunkVc) -> Self {
+        CssChunkSourceMapAssetReference { chunk }.cell()
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl AssetReference for CssChunkSourceMapAssetReference {
+    #[turbo_tasks::function]
+    async fn resolve_reference(&self) -> Result<ResolveResultVc> {
+        let source_maps = vec![CssChunkSourceMapAsset { chunk: self.chunk }.cell().into()];
+        Ok(ResolveResult::Alternatives(source_maps, vec![]).cell())
```

Comment:
The ES code allocates individual sourcemaps for each item, because they can be used as the maps during HMR. I'm not certain how CSS updates during dev, so this may be fine to exclude.

Reply:
It seems the hmr-client will reload the css chunk by [replace the link](https://github.com/vercel/turbo/blob/93222f2b22d43bd9c1d917350f810e5398ca9d49/crates/next-core/js/src/dev/hmr-client.ts#L320) for css updates during dev, so it might be fine to not allocates individual sourcemaps for css chunk items. (I'm not 100% sure)